core/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -54,8 +64,28 @@ public Task register(String type, String action, TransportRequest request) {
             if (logger.isTraceEnabled()) {
                 logger.trace("register {} [{}] [{}] [{}]", task.getId(), type, action, task.getDescription());
             }
+            try {
+                task.onRegistered();
+            } catch (Throwable t) {
+                logger.warn("failed to notify the task with id {} that it was registered", t, task.getId());
+            }
             Task previousTask = tasks.put(task.getId(), task);
             assert previousTask == null;
+            // Check if this task was banned before we start it
+            if (task.getParentNode() != null && banedParents.isEmpty() == false) {
+                String reason = banedParents.get(new Tuple<>(task.getParentNode(), task.getParentId()));
+                if (reason != null) {
+                    try {
+                        assert task instanceof CancellableTask;
```

Comment:
also here the class cast excep we get the line below is much more informative that the assert so we can skip the assert?